server/models/transactionModel.js

```diff
@@ -46,4 +47,50 @@ Transaction.init(
   }
 );
 
+// this hook will be called before creating a new transaction
+Transaction.addHook('afterCreate', async (transaction) => {
+  const budget = await Budget.findByPk(transaction.budgetId);
+
+  if (transaction.type === 'income') {
```

Comment:
use enum CategoryType

Reply:
added ENUMs in commit b65f80fff5815e341e0b7113865a636599e181a1